### PR TITLE
[SPIRV] Switch tiling on reductions to use common helpers.

### PR DESCRIPTION
**Pull request by hanhanW**

### CODE REVIEW

**Comment by hanhanW** [comment]
It's based on https://github.com/openxla/iree/pull/15939

File: compiler/src/iree/compiler/Codegen/SPIRV/SPIRVTile.cpp [hunk 1]
```diff
@@ -143,37 +146,34 @@ static LogicalResult tileAndDistributeToThreads(linalg::LinalgOp consumerOp,
...
-    setSCFTileSizes(options, target, tileSizes, /*tileScalableFlags=*/{});
-    FailureOr<scf::SCFTilingResult> tiledResults =
-        scf::tileUsingSCFForOp(rewriter, target, options);
-    if (failed(tiledResults)) {
-      return failure();
-    }
-    rewriter.replaceOp(op, tiledResults->replacements);
-  };
+  auto filter =
+      IREE::LinalgExt::LinalgTransformationFilter({marker}, std::nullopt);
```
Comment (hanhanW) [inline compiler/src/iree/compiler/Codegen/SPIRV/SPIRVTile.cpp]: This part is as same as https://github.com/openxla/iree/commit/9b74e956dafd5b11c5581943b4117bbc38eda115#diff-b8b284fc638e6a2c004e1c0102c7a41699d1a2b6bbc9d31b46ea79fdabd64b8aL183-L193

**Comment by antiagainst** [comment]
Why the revert? I think I'm missing context here..

**Comment by hanhanW** [comment]
> Why the revert? I think I'm missing context here..

oh I thought that we want to keep it consistent. Flipping it back and use common methods now. (and also for SPIRVTileAndDistribute)

**Review by antiagainst — APPROVED** [review]

**Pull request merged**